api/utils.go

```diff
@@ -52,6 +52,11 @@ func JSON(w http.ResponseWriter, v interface{}) {
 	JSONStatus(w, v, http.StatusOK)
 }
 
+// JSON writes the passed value into the http.ResponseWriter.
+func WriteProcessing(w http.ResponseWriter, v interface{}) {
+	JSONStatus(w, v, http.StatusProcessing)
```

Comment:
I don't think we want to use status processing right here. Specifically, https://tools.ietf.org/html/rfc2518#section-10.1 says:

> This status code SHOULD only be sent
> when the server has a reasonable expectation that the request will
> take significant time to complete. As guidance, if a method is taking
> longer than 20 seconds (a reasonable, but arbitrary value) to process
> the server SHOULD return a 102 (Processing) response. The server MUST
> send a final response after the request has been completed.

Also, in general HTTP is concerned with the exchange of information about resources not some domain-specific notion of resource state.